.claude/hooks/track.sh

```
@@ -2,56 +2,116 @@
+emit_hydrate() {
+  printf 'TASK-PLANNING HYDRATE: `$project hydrate`%s. Follow the task-planning skill "Project handoff" -> hydrate: identify the CURRENT project (the single `active` .agents/projects/registry.yml entry for the current user; if more than one, ask which), reconcile its TASKS.md (check off done, note next step on in-progress items), update its `resume:` field + the doc resume pointer, push decisions into its DESIGN.md and durable direction to memory, run git status and account for EVERY uncommitted file, then confirm the checkpoint in one short block (done / in-progress / next / uncommitted).\n' "$legacy_note"
+}
+
+emit_resume() {
+  printf 'TASK-PLANNING RESUME: `$project resume`%s. Follow the task-planning skill "Project handoff" -> resume: pick the project from .agents/projects/registry.yml — %s. Stay in this session'\''s assigned worktree: NEVER cd into, edit in, or adopt another project'\''s worktree or branch — if the prior work lives on an unmerged branch elsewhere, report that and ask the user instead of following it. Read the project'\''s `tasks` (TASKS.md) + `design` doc (memory is already loaded), check git status + recent git log, report a concise state (done / in-progress / next action / uncommitted), then continue with the next action unless the user directed otherwise.\n' \
+    "$legacy_note" "$1"
+}
+
+# --- unified `$project` sentinel ---------------------------------------------
+
+raw=$(printf '%s\n' "$prompt" | grep -ioE '\$project([[:space:]]+[^[:cntrl:]]*)?' | head -1 || true)
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)